src/components/MultiHopTrade/hooks/useAllowanceApproval/hooks/useExecuteAllowanceApproval.tsx

```diff
@@ -36,7 +36,11 @@ export const useExecuteAllowanceApproval = (
     }
 
     try {
-      const txId = await buildAndBroadcast({ buildCustomTxInput, adapter })
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: undefined, // no receiver for this contract call
```

Comment:
Wouldn't the "receiver" (as in, `to` address) be the contract here? 🤔

Reply:
Discussed this one out, not doing verifications for contracts for the time being